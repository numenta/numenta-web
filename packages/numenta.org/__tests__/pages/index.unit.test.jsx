import React from 'react'
import renderer from 'react-test-renderer'
import stubContext from 'react-stub-context'

import context from '__tests__/__mocks__/reactContextMock'

import IndexPage from '../../pages/index'


describe('Index Page React component', () => {

  it('Renders correctly', () => {
    const IndexPageStubbed = stubContext(IndexPage, context)
    const component = renderer.create(
      <IndexPageStubbed />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
