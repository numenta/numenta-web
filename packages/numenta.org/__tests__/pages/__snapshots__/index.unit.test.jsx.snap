exports[`Index Page React component Renders correctly 1`] = `
<div
  className="page">
  <div>
    <section
      className="section"
      id="sectionHome">
      <h1
        className="headline">
        Home of the HTM Community
      </h1>
      <div
        className="article">
        <article
          className="columns">
          <div
            className="aside">
            <aside
              className="glance">
              <div>
                <div
                  className="glanceHead">
                  Community Resources
                </div>
              </div>
              <dl
                className="defineList horizontal">
                <dt
                  className="defineTitle horizontal">
                  Discussion
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="https://discourse.numenta.org/categories"
                    onClick={[Function]}
                    target={undefined}>
                    HTM Forum
                  </a>
                   
                   
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Code
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="https://github.com/numenta"
                    onClick={[Function]}
                    target={undefined}>
                    Numenta Github
                  </a>
                  <br />
                  <a
                    className="textlink"
                    href="https://github.com/htm-community"
                    onClick={[Function]}
                    target={undefined}>
                    HTM Community Github
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Videos
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="https://www.youtube.com/c/NumentaOrg"
                    onClick={[Function]}
                    target={undefined}>
                    YouTube
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Technical
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="http://numenta.com/papers/"
                    onClick={[Function]}
                    target={undefined}>
                    Papers
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  License
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href={undefined}
                    onClick={[Function]}
                    style={Object {}}
                    target={undefined}>
                    AGPLv3
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Events
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="https://www.meetup.com/Numenta-Platform-for-Intelligent-Computing-Events/"
                    onClick={[Function]}
                    target={undefined}>
                    Meetup
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Blog
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href={undefined}
                    onClick={[Function]}
                    style={Object {}}
                    target={undefined}>
                    Blog
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  HTM School
                </dt>
                <dd
                  className="defineItem horizontal">
                  <p
                    className="paragraph">
                    See the latest episode below!
                  </p>
                  <div
                    className="noLinkIcon">
                    <a
                      className="imagelink"
                      href="https://www.youtube.com/watch?v=UBzemKcUoOk"
                      onClick={[Function]}
                      title={undefined}>
                      <img
                        alt="Latest HTM School Episode"
                        className="image border mw shadow"
                        onClick={undefined}
                        src="https://i3.ytimg.com/vi/UBzemKcUoOk/mqdefault.jpg"
                        title={undefined} />
                    </a>
                  </div>
                </dd>
              </dl>
            </aside>
          </div>
          <div
            className="content">
            <div
              className="markdown">
              <div
                className="markdownBody">
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Markdown Test Body Example",
                    }
                  } />
              </div>
            </div>
          </div>
        </article>
      </div>
    </section>
    <section
      className="section"
      id="sectionHtm">
      <nav
        className="title"
        onClick={[Function]}
        title="Open Section: Hierarchical Temporal Memory">
        <svg
          className="toggle"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m31 16.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.5 0.3t-0.6-0.3l-10.4-10.4q-0.2-0.2-0.2-0.5t0.2-0.5l1.2-1.1q0.2-0.2 0.5-0.2t0.5 0.2l8.8 8.8 8.7-8.8q0.3-0.2 0.5-0.2t0.6 0.2l1.1 1.1q0.2 0.2 0.2 0.5z" />
          </g>
        </svg>
        <h1
          className="title">
          Hierarchical Temporal Memory
        </h1>
      </nav>
      <div
        className="article closed">
        <article>
          <div
            className="columns">
            <div
              className="aside">
              <div
                className="markdownMedia" />
            </div>
            <div
              className="content">
              <div
                className="markdown">
                <div
                  className="markdownBody">
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Markdown Test Body Example",
                      }
                    } />
                </div>
              </div>
            </div>
          </div>
        </article>
        <div
          className="closeButton"
          onClick={[Function]}
          title="Close Section: Hierarchical Temporal Memory">
          <svg
            className="closeIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "inherit",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em">
            <g>
              <path
                d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z" />
            </g>
          </svg>
          Close Section
        </div>
      </div>
    </section>
    <section
      className="section"
      id="sectionCommunity">
      <nav
        className="title"
        onClick={[Function]}
        title="Open Section: Community">
        <svg
          className="toggle"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m31 16.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.5 0.3t-0.6-0.3l-10.4-10.4q-0.2-0.2-0.2-0.5t0.2-0.5l1.2-1.1q0.2-0.2 0.5-0.2t0.5 0.2l8.8 8.8 8.7-8.8q0.3-0.2 0.5-0.2t0.6 0.2l1.1 1.1q0.2 0.2 0.2 0.5z" />
          </g>
        </svg>
        <h1
          className="title">
          Community
        </h1>
      </nav>
      <div
        className="article closed">
        <article>
          <div
            className="columns">
            <div
              className="aside">
              <img
                alt="HTM Community Event"
                className="image border mw shadow"
                onClick={undefined}
                src="filename.ext"
                title={undefined} />
            </div>
            <div
              className="content">
              <p
                className="paragraph lead">
                Our community is an eclectic collection of researchers, scientists, hobbyists, and hackers interested in building biologically-inspired intelligent systems with Hierarchical Temporal Memory (HTM).
              </p>
              <p
                className="paragraph">
                Sponsored by
                <a
                  className="textlink"
                  href="http://numenta.com/"
                  onClick={[Function]}
                  target={undefined}>
                   
                   
                  Numenta
                </a>
                 
                 
                and supported by a full-time employee dedicated to fostering and growing the community, we strive to create an inclusive environment. Our community exemplifies the type of people drawn toward new technology, and it shows in our curiosity and open-mindedness.
              </p>
              <p
                className="paragraph">
                To get involved in our community, join 
                 
                <a
                  className="textlink"
                  href="https://discourse.numenta.org/categories"
                  onClick={[Function]}
                  target={undefined}>
                  HTM Forum.
                </a>
                 
                 
                You can login with your Google, Facebook, or Twitter account, or by creating a new account with your email address. This is the best place to ask questions, search for answers, or just interact with others working on similar problems. We aim to be a helpful and welcoming community, and we hope you\'ll join us.
              </p>
            </div>
          </div>
          <span
            className="anchor"
            id="events" />
          <h2
            className="h2">
            Meetups, Hackathons, & Other Live Events
          </h2>
          <div
            className="columns">
            <div
              className="aside">
              <img
                alt="HTM Community Event #2"
                className="image border mw shadow"
                onClick={undefined}
                src="filename.ext"
                title={undefined} />
            </div>
            <div
              className="content">
              <p
                className="paragraph">
                While many of our interactions are virtual, we do hold community events like meetups, hackathons, and workshops throughout the year. All our live events are scheduled via Meetup. Please join our 
                 
                <a
                  className="textlink"
                  href="https://www.meetup.com/Numenta-Platform-for-Intelligent-Computing-Events/"
                  onClick={[Function]}
                  target={undefined}>
                  Meetup Group
                </a>
                 
                 
                for notifications of upcoming events, all of which are free and open to anyone. Please read our 
                 
                <a
                  className="textlink"
                  href={undefined}
                  onClick={[Function]}
                  style={Object {}}
                  target={undefined}>
                  code of conduct
                </a>
                 
                 
                before attending events.
              </p>
              <p
                className="paragraph">
                For recordings of previous events, please see our 
                 
                <a
                  className="textlink"
                  href="https://www.youtube.com/c/NumentaOrg"
                  onClick={[Function]}
                  target={undefined}>
                  YouTube channel.
                </a>
                 
                 
                You may also be interested in Numenta\'s 
                 
                <a
                  className="textlink"
                  href="http://numenta.com/events/"
                  onClick={[Function]}
                  target={undefined}>
                  company events and archive.
                </a>
              </p>
            </div>
          </div>
        </article>
        <div
          className="closeButton"
          onClick={[Function]}
          title="Close Section: Community">
          <svg
            className="closeIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "inherit",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em">
            <g>
              <path
                d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z" />
            </g>
          </svg>
          Close Section
        </div>
      </div>
    </section>
    <section
      className="section"
      id="sectionCode">
      <nav
        className="title"
        onClick={[Function]}
        title="Open Section: Code">
        <svg
          className="toggle"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m31 16.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.5 0.3t-0.6-0.3l-10.4-10.4q-0.2-0.2-0.2-0.5t0.2-0.5l1.2-1.1q0.2-0.2 0.5-0.2t0.5 0.2l8.8 8.8 8.7-8.8q0.3-0.2 0.5-0.2t0.6 0.2l1.1 1.1q0.2 0.2 0.2 0.5z" />
          </g>
        </svg>
        <h1
          className="title">
          Code
        </h1>
      </nav>
      <div
        className="article closed">
        <article>
          <div
            className="columns">
            <div
              className="aside">
              <div
                className="markdownMedia" />
            </div>
            <div
              className="content">
              <div
                className="markdown">
                <div
                  className="markdownBody">
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Markdown Test Body Example",
                      }
                    } />
                </div>
              </div>
            </div>
          </div>
        </article>
        <div
          className="closeButton"
          onClick={[Function]}
          title="Close Section: Code">
          <svg
            className="closeIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "inherit",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em">
            <g>
              <path
                d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z" />
            </g>
          </svg>
          Close Section
        </div>
      </div>
    </section>
  </div>
</div>
`;
